proposals/0035-linalg-matrix.md

```diff
@@ -262,6 +287,26 @@ Throughout this document a matrix may be described as having a scope as
 specified by the `Scope` parameter (e.g. a matrix with `Scope == Thread` is a
 _matrix with thread scope_).
 
+Matrix storage is always opaque, the `Scope` does not directly restrict how the
+matrix is stored, it merely denotes allowed scopes of allowed data divergence.
+A matrix with thread scope must behave as if each thread has a unique copy of
+the matrix, however the storage is not required to be thread local, and an
+implementation may coalesce identical matrices across threads.
+
+#### Matrix Storage
+
+In HLSL, matrix objects are intangible objects so they do not have defined size
+or memory layout. When in use, implementations are expected to spread the
+storage of matrices across the thread-local storage for all threads in a SIMD
+unit. At the DXIL level we represent a matrix through a handle object.

```

Comment:
Are these two statements compatible?

> the storage is not required to be thread local, and an implementation may coalesce identical matrices across threads.

> implementations are expected to spread the storage of matrices across the thread-local storage for all threads in a SIMD unit

Reply:
Yes, although probably confusingly worded...

The point is that the _full_ storage for a matrix does not need to be materialized into a single thread, instead an implementation is expected to spread storage of matrices across thread-local storage for all threads in a SIMD unit.